test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/AddressControllerSpec.scala

```diff
@@ -238,14 +239,79 @@ trait AddressControllerSpec[A <: AddressJourneyType]
       }
     }
 
+    "redirect to the exit page" when {
+      "a registered without id user attempts to change their address to a UK address" in {
+        validJourneyStatus match {
+          case j: AddressJourneyType.ManagingSubscription.SubscribedAddressJourney =>
+            val subscribedDetails: SubscribedDetails = sample[SubscribedDetails].copy(registeredWithId = false)
```

Comment:
This is all smelly - we have to leak the exact JournetStatus type into the AddressControllerSpec but I can't see any other way to do it